source/manual/query-cdn-logs.html.md

```diff
@@ -183,6 +183,253 @@ ORDER BY count DESC
+    ROUND(100.0 / COUNT(*) * SUM(CASE WHEN status / 100 = 3 THEN 1 ELSE 0 END), 3) AS s3xx_pc,
+                             SUM(CASE WHEN status / 100 = 4 THEN 1 ELSE 0 END)     AS s4xx,
+    ROUND(100.0 / COUNT(*) * SUM(CASE WHEN status / 100 = 4 THEN 1 ELSE 0 END), 3) AS s4xx_pc,
+                             SUM(CASE WHEN status / 100 = 5 THEN 1 ELSE 0 END)     AS s5xx,
+    ROUND(100.0 / COUNT(*) * SUM(CASE WHEN status / 100 = 5 THEN 1 ELSE 0 END), 3) AS s5xx_pc,
+                             SUM(CASE WHEN cache_response NOT IN ('PASS', 'MISS') THEN 1 ELSE 0 END)     AS cache_hits,
+    ROUND(100.0 / COUNT(*) * SUM(CASE WHEN cache_response NOT IN ('PASS', 'MISS') THEN 1 ELSE 0 END), 3) AS cache_hit_pc,
+                             SUM(CASE WHEN cache_response IN ('PASS', 'MISS') THEN 1 ELSE 0 END)         AS cache_misses,
+    ROUND(100.0 / COUNT(*) * SUM(CASE WHEN cache_response IN ('PASS', 'MISS') THEN 1 ELSE 0 END), 3)     AS cache_miss_pc
+FROM fastly_logs.govuk_www
+WHERE year = 2021 AND month = 1 AND date = X
+GROUP BY 1
+ORDER BY 1 ASC;
+```
+
+### Minutes during the week where error rate was >0.1%
+
+This is the "availability" metric shown on the SMT dashboard.
+
+```sql
+SELECT
+  date_trunc('week', minute) AS week,
+  COUNT(*) AS mins_with_bad_error_rate,
+  -- Next line assumes we're calculating for a full week (7 days * 24 hours * 60 minutes)
+  -- This will need to be updated if the calculation range is changed.
+  -- It'll also be off during weeks where the clocks go forwards / back.
+  1.0 - (1.0 / (7 * 24 * 60) * COUNT(*)) AS availability
+FROM (
+  SELECT
+    date_trunc('minute', request_received) AS minute,
+    COUNT(*) AS total_requests,
+    SUM(CASE WHEN status BETWEEN 200 AND 499 THEN 1 ELSE 0 END) AS successful_requests,
+    100.0 / COUNT(*) * SUM(CASE WHEN status BETWEEN 200 AND 499 THEN 1 ELSE 0 END) AS availability
+  FROM fastly_logs.govuk_www
+  WHERE year = 2021 AND month = 1 AND date >= X AND date <= Y
+    AND user_agent NOT LIKE 'GOV.UK Crawler Worker%'
+    AND request_received > TIMESTAMP '2021-01-12 15:00:00'
+  GROUP BY date_trunc('minute', request_received)
+)
+WHERE availability < 99.9
+GROUP BY date_trunc('week', minute)
+ORDER BY date_trunc('week', minute) ASC;
+```
+
+## Snippets
+
+Little snippets of bits of SQL.
+
+### Path without query string
+
+```sql
+SPLIT_PART(url, '?', 1) AS url_path
+```
+
+### Exclude the GOV.UK Crawler
+
+```sql
+user_agent NOT LIKE 'GOV.UK Crawler Worker%'
+```
+
+### Remove extra mirror path information
+
+If an origin request fails and the request is failed over to a static mirror,
+the path is changed based on the location of the static file within the S3
+bucket. E.g. `/some/path` --> `/www.gov.uk/some/path.html`
+
+```sql
+REPLACE(REPLACE(SPLIT_PART(url, '?', 1), '/www.gov.uk', ''), '.html', '') AS url_path
+```
+
+### Count cases of a thing
+
+```sql
+SUM(CASE WHEN ... THEN 1 ELSE 0 END)
```

Comment:
Pro-tip for if you do more presto queries, they have a [COUNT_IF](https://docs.data.world/documentation/sql/reference/aggregations/count_if.html) function to simplify these scenarios.